jdbc-example/src/main/java/com/example/dao/Main.java

```diff
@@ -42,6 +50,23 @@ public static void main(String[] args) {
                 }
                 ConfigurationProvider configClient = new CortexClient(endpoint, token, project);
                 ConfigurationProvider.JdbcConnectionParams jdbcParams = configClient.getConnectionParams(connectionName);
+                // validate params
+                LOGGER.info(String.format("Connection properties: %s", jdbcParams.getConnectionProperties().keySet()));
+                List<String> errors = configClient.validateConnectionParams(jdbcParams);
+                if (!errors.isEmpty()) {
+                    errors.forEach(e -> LOGGER.error(e));
+                    throw new Exception(String.format("Connection parameters validation failed. %s", errors));
+                }
+                // check any secure files like service account volume mapped
+                Files.list(Paths.get("/secure-storage")).forEach(f -> {
```

Comment:
Shouldn't this be in the below try catch? What happens if my pod spec isn't loaded correctly causing secure-storage not being available in the path?

Reply:
handled exception for optional mounts